homeassistant/components/hive/sensor.py

```diff
@@ -1,20 +1,50 @@
 """Support for the Hive sensors."""
 from datetime import timedelta
 
-from homeassistant.components.sensor import SensorDeviceClass, SensorEntity
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+)
 from homeassistant.config_entries import ConfigEntry
 from homeassistant.core import HomeAssistant
-from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.entity import DeviceInfo, EntityCategory
 from homeassistant.helpers.entity_platform import AddEntitiesCallback
 
 from . import HiveEntity
 from .const import DOMAIN
 
 PARALLEL_UPDATES = 0
 SCAN_INTERVAL = timedelta(seconds=15)
-DEVICETYPE = {
-    "Battery": {"unit": " % ", "type": SensorDeviceClass.BATTERY},
-}
+
+
+SENSOR_TYPES: tuple[SensorEntityDescription, ...] = (
+    SensorEntityDescription(key="Heating_Mode", icon="mdi:radiator"),
+    SensorEntityDescription(key="Hotwater_Mode", icon="mdi:water-pump"),
+    SensorEntityDescription(key="Hotwater_Boost", icon="mdi:water-pump"),
+    SensorEntityDescription(key="Mode", icon="mdi:eye"),
+    SensorEntityDescription(
+        key="Battery",
+        native_unit_of_measurement="%",
+        device_class=SensorDeviceClass.BATTERY,
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
+    SensorEntityDescription(
+        key="Availability",
+        icon="mdi:check-circle",
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
+    SensorEntityDescription(
+        key="Connectivity",
+        icon="mdi:check-circle",
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
+    SensorEntityDescription(
+        key="Power",
+        device_class=SensorDeviceClass.ENERGY,
```

Comment:
We're missing a unit of measurement here. The energy device class must use an energy unit according to our dev docs:
https://developers.home-assistant.io/docs/core/entity/sensor#available-device-classes